### PR TITLE
feat(security): improve general security of the website

### PR DESCRIPTION
**Pull request by Huskydog9988**
## **Description**

- [x] stricter sanitization on the server
- [x] perms check on all endpoints
- [x] sanitize data again before displaying in website (react serves our needs already)
- [x] #98 
- [x] #96 via #129

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "eslint": "8.29.0",
     "eslint-config-next": "13.0.6",
     "framer-motion": "^7.6.19",
+    "html-entities": "^2.3.3",
```
Comment (Huskydog9988) [inline apps/web/package.json:34]: why was this added @ceriddenn?

**Review by tsundDEV — APPROVED** [review]

**Review by jaredb1011 — APPROVED** [review]
